### PR TITLE
Feature/create login successfully

### PR DESCRIPTION
**Pull request by DKnehi**
- Implement function login successfully

### CODE REVIEW

File: javascript-practice/src/js/constants/message.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const VALIDATE_ERORR = {
+  Required_Field: 'This field cannot be empty!',
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/constants/message.js]: Not the same format variable between constant file. Pls check all

File: javascript-practice/src/js/controllers/userController.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    this.service = new UserService()
+    
+    this.view.bindFormLogin(this.handleFormLogin);
+  }
+
+  handleFormLogin = async (email, password) => {
+    // call api get 
+    try {
+      const userData = await this.service.checkEmailPassword(email, password);
+      console.log(userData);
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/controllers/userController.js]: Remove console.log

File: javascript-practice/src/js/controllers/userController.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      if(userData) {
+        this.view.redirectPage("dashboard.html");
+      }
+      else {
+        alert('Đăng nhập ko thành công')
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/controllers/userController.js]: Avoid vnese text

File: javascript-practice/src/js/services/userService.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+          });
+          if (res.ok) {
+            return await res.json();
+          }
+        } catch (error) {
+          throw new Error(`Get data fail ${error.message}`);
+        }
+      };
+
+      checkEmailPassword = async (email, password) => {
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/services/userService.js]: Not a good name. Should clear what it does

File: javascript-practice/src/js/services/userService.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        try {
+            const data = await this.getAllUser();
+            const user = data.find(
+              (user) => user.email === email && user.password === password
+            );
+            return user;
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/services/userService.js]: After const, should have empty line

File: javascript-practice/src/js/views/userView.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+export default class UserView {
+  constructor() {
+    this.emailData = document.getElementById('email');
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/views/userView.js]: Consider another name, should clear what it is. For now, you get ELEMENT but name is DATA?

Btw, why define this as class variable but not another scopes?

File: javascript-practice/src/js/helpers/validation.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+const { Required_Field, Invalid_Email, Invalid_Password } = VALIDATE_ERORR
+
+const { EmailRegex, PasswordRegex } = REGEX
+
+const isRequired = value => (value !== '' ? '' : Required_Field);
+const isEmail = value => EmailRegex.test(value) ? '' : Invalid_Email;
+const isPassword = value => PasswordRegex.test(value) ? '' : Invalid_Password;
+
+
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/helpers/validation.js]: Remove redundant empty line. One is enough

File: javascript-practice/src/js/helpers/validation.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import { VALIDATE_ERORR } from '../constants/message'
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/helpers/validation.js:1]: Where do you use it?

File: javascript-practice/src/pages/dashboard.html [hunk 9]
```diff
@@ -6,6 +6,6 @@
   <title>Dashboard</title>
 </head>
 <body>
-  
+  <h2>DashBoard</h2>
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/pages/dashboard.html]: Typo

File: javascript-practice/src/pages/index.html [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
+</head>
+<body>
+    <h2>Index</h2>
+</body>
+</html>
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/pages/index.html]: Need empty line at the end of file

Comment (phuocduong-agilityio) [reply javascript-practice/src/pages/index.html]: Check format file indent, indent style should be 2 spaces for all files

File: javascript-practice/src/styles/main.css [hunk 11]
```diff
@@ -3,6 +3,7 @@
 @import "./components/index.css";
 @import "./layouts/index.css";
 @import "./pages/login.css";
+@import "./modal/index.css";
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/styles/main.css]: Why do you have folder for modal? Should we use it as components?

Comment (phuocduong-agilityio) [reply javascript-practice/src/styles/main.css]: Btw, I found you have NO CODE for modal. Should not add empty file until you use it.

File: javascript-practice/src/js/controllers/userController.js [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+  constructor() {
+    this.view = new UserView()
+    this.model = new UserModel()
+    this.service = new UserService()
+    
+    this.view.bindFormLogin(this.handleFormLogin);
+  }
+
+  handleFormLogin = async (email, password) => {
+    // call api get
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/controllers/userController.js]: Not right format for comment code, please check https://www.digitalocean.com/community/tutorials/how-to-write-comments-in-javascript

File: javascript-practice/src/js/constants/message.js (hunk 1 above)
Comment (DKnehi) [reply javascript-practice/src/js/constants/message.js]: I done fix it

File: javascript-practice/src/js/controllers/userController.js (hunk 2 above)
Comment (DKnehi) [reply javascript-practice/src/js/controllers/userController.js]: I done fix it

File: javascript-practice/src/js/constants/message.js [hunk 13]
```diff
@@ -0,0 +1,12 @@
+const VALIDATE_ERORR = {
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/constants/message.js]: Typo

File: javascript-practice/src/js/constants/message.js [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+  REQUIRD_FIELD: 'This field cannot be empty!',
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/constants/message.js]: Typo

File: javascript-practice/src/js/constants/message.js [hunk 15]
```diff
@@ -0,0 +1,12 @@
...
+  INVALID_EMAIL: 'Invalid email, please re-enter!',
+  INVALID_PASSWORD: 'Password must be more than 6 characters, including uppercase letters and special characters!',
+}
+
+const NOTIFY_MESSAGE = {
+  INVALID_LOGIN_FAILED: 'Login failed. Please check your login information again!',
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/constants/message.js]: Invalid or failed/success?

File: javascript-practice/src/js/controllers/userController.js [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+    this.model = new UserModel()
+    this.service = new UserService()
+
+    this.view.bindFormLogin(this.handleFormLogin);
+  }
+
+  handleFormLogin = async (email, password) => {
+    try {
+      const userData = await this.service.loginUser(email, password);
+      if (userData) {
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/controllers/userController.js:17]: Update code style: https://github.com/airbnb/javascript?tab=readme-ov-file#blocks--cuddled-elses

File: javascript-practice/src/js/controllers/userController.js [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+    this.view = new UserView()
+    this.model = new UserModel()
+    this.service = new UserService()
+
+    this.view.bindFormLogin(this.handleFormLogin);
+  }
+
+  handleFormLogin = async (email, password) => {
+    try {
+      const userData = await this.service.loginUser(email, password);
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/controllers/userController.js:16]: After const should have empty line.

File: javascript-practice/src/js/main.js [hunk 18]
```diff
@@ -0,0 +1,11 @@
+// import UserView from "./views/userView";
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/main.js]: Remove code if you don't use it. Btw, should have only 1 empty line at the end of file

File: javascript-practice/src/js/services/userService.js [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+    }
+  };
+
+  loginUser = async (email, password) => {
+    try {
+      const data = await this.getAllUser();
+      const user = data.find(
+        (user) => user.email === email && user.password === password
+      );
+      return user;
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/services/userService.js:24]: Need empty line before return

File: javascript-practice/src/js/views/userView.js [hunk 20]
```diff
@@ -0,0 +1,20 @@
+export default class UserView {
+  constructor() {
+    this.emailEl = document.getElementById('email');
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/views/userView.js]: Why we need define emailEl as class variable?

File: javascript-practice/src/js/views/userView.js [hunk 21]
```diff
@@ -0,0 +1,20 @@
...
+    this.passwordEl = document.getElementById('password');
+    this.loginFormEl = document.getElementById('login-form');
+  }
+
+  bindFormLogin = (handler) => {
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/views/userView.js]: Should use more specific name than handler. Handler is too common

File: javascript-practice/src/styles/components/index.css [hunk 22]
```diff
@@ -1,2 +1,3 @@
 @import "./button.css";
 @import "./input.css";
+@import "./error.css";
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/styles/components/index.css]: Please remove if you don't use it

File: javascript-practice/src/js/services/userService.js (hunk 19 above)
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/services/userService.js:25]: @DKnehi Before not after

File: javascript-practice/src/js/views/userView.js [hunk 23]
```diff
@@ -0,0 +1,18 @@
+export default class UserView {
+  constructor() {
+    this.loginFormEl = document.getElementById('login-form');
+  }
+
+  bindFormLogin = (submitCallBack) => {
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/views/userView.js]: submitCallback (not submitCallBack) is not clear what we do, consider another name.

File: javascript-practice/src/pages/dashboard.html (hunk 9 above)
Comment (phuocduong-agilityio) [reply javascript-practice/src/pages/dashboard.html]: @DKnehi Pls check

File: javascript-practice/src/js/views/userView.js [hunk 24]
```diff
@@ -0,0 +1,18 @@
+export default class UserView {
+  constructor() {
+    this.loginFormEl = document.getElementById('login-form');
+  }
+
+  bindFormLogin = (submitLogin) => {
+    this.loginFormEl.addEventListener("submit", e => {
+      e.preventDefault();
+      const valueEmail = document.getElementById('email').value;
+      const valuePassword = document.getElementById('password').value;
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/views/userView.js:10]: Add empty line after const

File: javascript-practice/src/js/constants/regex.js [hunk 25]
```diff
@@ -0,0 +1,6 @@
+export const REGEX = {
+  REQUIRD_FIELD_REGEX: /^.*\S+.*$/,
```
Comment (hoangle-agilityio) [inline javascript-practice/src/js/constants/regex.js]: Remove `REGEX` from your naming. Because name of object is REGEX. That's enough

**Pull request merged**